posts/ml-sagemaker.md

```diff
@@ -0,0 +1,201 @@
+
+```
+
+
+Fetch the container with proper algorithm to use from the list of pre-defined Sagemaker algorithms or give your own custom container to support custom algorithms. In this case linear-learner algorithm is used which is a pre-defined algorithm.
+
+``` python
+
+from sagemaker.amazon.amazon_estimator import get_image_uri
+linear_container = get_image_uri(boto3.Session().region_name, 'linear-learner')
+
+```
+
+
+Now train the Model using the container and the training data previously prepared. Create a new instance for training the Model, provide the instance type needed. The trained Model is stored in the S3 bucket as a tar file so provide S3 bucket details.
+
+Note: Only certain types of instance types can be used for training and deploying the models. You will be warned with a message as below.
+``` text
+ClientError: An error occurred (ValidationException) when calling the CreateTrainingJob operation: 
+1 validation error detected: Value 'ml.t2.medium' at 'resourceConfig.instanceType' failed to 
+satisfy constraint: Member must satisfy enum value set: [ml.p2.xlarge, ml.m5.4xlarge, 
+ml.m4.16xlarge, ml.p3.16xlarge, ml.m5.large, ml.p2.16xlarge, ml.c4.2xlarge, ml.c5.2xlarge, 
+ml.c4.4xlarge, ml.c5.4xlarge, ml.c4.8xlarge, ml.c5.9xlarge, ml.c5.xlarge, ml.c4.xlarge,
+ ml.c5.18xlarge, ml.p3.2xlarge, ml.m5.xlarge, ml.m4.10xlarge, ml.m5.12xlarge, ml.m4.xlarge,
+ml.m5.24xlarge, ml.m4.2xlarge, ml.p2.8xlarge, ml.m5.2xlarge, ml.p3.8xlarge, ml.m4.4xlarge]
+```
+
+``` python
+from sagemaker import get_execution_role
+
+role = get_execution_role()
+
+sagemaker_session = sagemaker.Session()
+
+// Provide the container, role, instance type and model output location
+linear = sagemaker.estimator.Estimator(linear_container,
+                                       role=role, 
+                                       train_instance_count=1, 
+                                       train_instance_type='ml.c4.xlarge',
+                                       output_path=output_location,
+                                       sagemaker_session=sagemaker_session)
+
+// Provide the number of features identified during data preparation
+// Provide the predictor_type 
+
+linear.set_hyperparameters(feature_dim=14,
+                           mini_batch_size=4,
+                           predictor_type='regressor')
+
+// Train the model using the previously prepared test data and validate the 
+//data by providing the validation data.
+
+linear.fit({'train': s3_training_data_location})
+
+```
+
+
+Deploy the Trained Model using the Sagemaker API. Provide instance type and instance count as required. Once the deployment is complete the test data is used to test the deployed application. Once tha Model is deployed a http endpoint is generated which is used by other applications such as a lambda function as part of a streaming application or a synchronous application. To support varying  loads the application can be autoscaled.
```

Comment:
```suggestion
Deploy the Trained Model using the Sagemaker API. Provide the instance type and instance count as required. Once the deployment is complete, the test data is used to test the deployed application. Once the Model is deployed an http endpoint is generated which is used by other applications such as a lambda function as part of a streaming application or a synchronous application. To support varying  loads the application can be autoscaled.
```
where do you find the Sagemaker API?

Did you mean _a synchronous_ or did you mean asynchronous?